moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -831,5 +831,52 @@ bool JointModelGroup::isValidVelocityMove(const double* from_joint_pose, const d
 
   return true;
 }
+
+std::pair<Eigen::VectorXd, Eigen::VectorXd> JointModelGroup::getLowerAndUpperLimits()
+{
+  // Get the group joints lower/upper position limits.
+  Eigen::VectorXd lower_limits(active_variable_count_);
+  Eigen::VectorXd upper_limits(active_variable_count_);
+  int joint_index = 0;
+  for (const moveit::core::JointModel::Bounds* joint_bounds : active_joint_models_bounds_)
+  {
+    for (const moveit::core::VariableBounds& variable_bounds : *joint_bounds)
+    {
+      lower_limits[joint_index] = variable_bounds.min_position_;
+      upper_limits[joint_index] = variable_bounds.max_position_;
+      joint_index++;
+    }
+  }
+  return { lower_limits, upper_limits };
+}
+
+std::pair<Eigen::VectorXd, Eigen::VectorXd> JointModelGroup::getMaxVelocitiesAndAccelerationBounds()
+{
+  Eigen::VectorXd max_joint_velocities = Eigen::VectorXd::Constant(active_variable_count_, 0.0);
+  Eigen::VectorXd max_joint_accelerations = Eigen::VectorXd::Constant(active_variable_count_, 0.0);
+  if (active_joint_models_bounds_.size() != active_variable_count_)
+  {
+    // TODO(sjahr) Support multiple variables
+    RCLCPP_ERROR(getLogger(), "Number of active joint models does not match number of active joint model bounds. "
+                              "Returning bound vectors.with zeros");
+    return { max_joint_velocities, max_joint_accelerations };
+  }
+  for (std::size_t i = 0; i < active_joint_models_bounds_.size(); ++i)
+  {
+    if (active_joint_models_bounds_[i]->size() == 1)
+    {
+      RCLCPP_ERROR(
+          getLogger(),
+          "Joint '%li' has '%li' bounds and only joints with one bounds set are supported by this function. In "
+          "the returned vector this element will be 0.",
```

Comment:
Is this true? will it be zero or will it be just not included? Maybe catch this case with the check above and return all zeros?